src/net.cpp

```diff
@@ -1693,6 +1693,38 @@ void CConnman::ProcessOneShot()
     }
 }
 
+bool CConnman::GetTryNewOutboundPeer()
+{
+    LOCK(m_cs_outbound_peer);
+    return m_try_another_outbound_peer;
+}
+
+void CConnman::SetTryNewOutboundPeer(bool flag)
+{
+    LOCK(m_cs_outbound_peer);
+    m_try_another_outbound_peer = flag;
+}
+
+// Return the number of peers we have over our outbound connection limit
+// Exclude peers that are marked for disconnect, or are going to be
+// disconnected soon (eg one-shots and feelers)
+// Also exclude peers that haven't finished initial connection handshake yet
+// (so that we don't decide we're over our desired connection limit, and then
+// evict some peer that has finished the handshake)
+int CConnman::GetExtraOutboundCount()
```

Comment:
I think this might be greatly simplified by extending CSemaphore to have an adjustable threshold.

Reply:
Any thoughts on @TheBlueMatt's suggestion (mentioned on IRC yesterday) to just use the feeler connection for the extra peer, rather than initiate a potential 10th connection?

https://botbot.me/freenode/bitcoin-core-dev/2017-10-27/?msg=92828477&page=2

I'm leaning towards doing that, so nMaxExtraOutbound would go away.  I guess this function wouldn't change though.